homeassistant/components/feedreader/config_flow.py

```diff
@@ -0,0 +1,137 @@
+"""Config flow for RSS/Atom feeds."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any
+import urllib.error
+
+import feedparser
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import CONF_MAX_ENTRIES, DEFAULT_MAX_ENTRIES, DOMAIN
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def async_fetch_feed(hass: HomeAssistant, url: str) -> feedparser.FeedParserDict:
+    """Fetch the feed."""
+    return await hass.async_add_executor_job(feedparser.parse, url)
+
+
+class FlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    _config_entry: ConfigEntry
+
+    def show_user_form(
+        self,
+        user_input: dict[str, Any] | None = None,
+        errors: dict[str, str] | None = None,
+        description_placeholders: dict[str, str] | None = None,
+        step_id: str = "user",
+    ) -> ConfigFlowResult:
+        """Show the user form."""
+        if user_input is None:
+            user_input = {}
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_URL, default=user_input.get(CONF_URL, "")
+                    ): TextSelector(TextSelectorConfig(type=TextSelectorType.URL)),
+                    vol.Optional(
+                        CONF_MAX_ENTRIES,
+                        default=user_input.get(CONF_MAX_ENTRIES, DEFAULT_MAX_ENTRIES),
+                    ): cv.positive_int,
```

Comment:
I am also wondering, wouldn't this be something for just an options flow? Why did we choose to put max entries in data and use reconfigure to change that?

Reply:
Because it is a parameter to be set during setup, it should be changeable via re-configure flow instead of options flow. But I'm not 100% sure when to use which one (_options and/or re-configure flow_)

Reply:
I'm still on the side that the re-configure flow is the better choice, since the feedreader has exactly two arguments - the URL and this "max entries" - both are changeable via the re-configure flow. if we would move the "max entries" in an options flow, than the user has two possibilities to change integration related paramaters, each time only one parameter ... IMO from ux perspective this feels odd :thinking:

Reply:
Mind if we double check this with a second opinion?

Reply:
sure :+1:

Reply:
From an UX perspective I concur with @mib1185 . But I understand that this might be against the intended separation of option flow.

Reply:
We want to ask as little as possible in our data flows, so I agree with joostlek.

Let's set a sane default and move it into an option flow.